patches/tModLoader/Terraria/ModLoader/IO/TileIO_TEs.cs

```diff
@@ -65,7 +66,19 @@ internal static partial class TileIO
 					}
 				}
 				//Otherwise, if the TE is vanilla, try to find its existing instance for the current coordinate.
-				else if (!TileEntity.ByPosition.TryGetValue(point, out tileEntity)) {
+				else if (TileEntity.ByPosition.TryGetValue(point, out tileEntity)) {
```

Comment:
hm... I don't really like this but I don't have a great solution right now. Perhaps we should make all modded pylons use a common, tML provided, modded TE?

Yes, I think you have to do this, because there's also 'unloaded tiles' to consider. With modded tiles and tile entities, pylons should survive a world load without the mod enabled.